components/Budget/ViewBudgetModal.jsx

```diff
@@ -0,0 +1,58 @@
+import { useState, useEffect, useContext } from "react";
+import { View, Text, ScrollView } from "react-native";
+import { ref, onValue } from "firebase/database";
+import { AlertContext } from "../Alert/AlertProvider";
+import { ModalContext } from "../Modal/ModalProvider";
+import { firebase_auth, firebase_db } from "../../firebase.config";
+import BudgetInfo from "./BudgetInfo.jsx";
+import { icon } from "@fortawesome/fontawesome-svg-core/import.macro";
+
+export default function ViewBudgetModal({ productionCode }) {
+    const { setAlert } = useContext(AlertContext);
+    //const { setModal } = useContext(ModalContext);
+    const [budgets, setBudgets] = useState([]);
+    //const auth = firebase_auth;
+    const db = firebase_db;
+
+    useEffect(() => {
+        try {
+            onValue(ref(db, `productions/${productionCode}/budgets`), (productionSnapshot) => {
+                if (!productionSnapshot.exists()) {
+                    return;
+                }
+
+                setBudgets(Object.keys(productionSnapshot.val()));
+            });
+        } catch (error) {
+            setAlert(
+                "Error occurred when fetching budgets.",
+                "bg-red-500",
+                icon({ name: "circle-exclamation" })
+            );
+            return;
+        }
+    }, []);
+
+    return (
+        <View className="flex p-3">
+            <Text className="text-3xl font-extrabold text-center mb-3">Select Budget</Text>
+            <ScrollView className="flex flex-col h-96">
+                {budgets.length > 0 ? (
+                    budgets.map((budget, i) => {
+                        return (
+                            <BudgetInfo
```

Comment:
Think I put the comment in the wrong place earlier with the having a permanent loading, basically getting that there is a loading screen permanently and it's not obvious to the user as to why